mautic/mautic
# Autowiring controllers.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        |  [ x ]
| Automated tests included?              | [ x ] 



#### Description:
The PR adds Controllers to the DI and allows autowiring of services. This is a step towards making all necessary services private, thus allowing to isolate parts of the Mautic and enabling symfony compiler to inline some of the services cutting down runtime costs.


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Check every possible part of the system.
